x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/LogicalPlanBuilder.java

```diff
@@ -160,9 +161,37 @@ else if (!selectTarget.isEmpty()) {
     public LogicalPlan visitFromClause(FromClauseContext ctx) {
         // if there are multiple FROM clauses, convert each pair in a inner join
         List<LogicalPlan> plans = plans(ctx.relation());
-        return plans.stream()
+        LogicalPlan plan = plans.stream()
                 .reduce((left, right) -> new Join(source(ctx), left, right, Join.JoinType.IMPLICIT, null))
                 .get();
+
+        // PIVOT
+        if (ctx.pivotClause() != null) {
+            PivotClauseContext pivotClause = ctx.pivotClause();
+            UnresolvedAttribute column = new UnresolvedAttribute(source(pivotClause.column), visitQualifiedName(pivotClause.column));
+            List<NamedExpression> values = namedValues(pivotClause.aggs);
+            if (values.size() > 1) {
+                throw new ParsingException(source(pivotClause.aggs), "PIVOT currently supports only one aggregation, found [{}]",
```

Comment:
I think `ParsingException` at this point is "late", but on the other hand `SqlIllegalArgumentException` doesn't seem appropriate. Should we introduce a new one, something with 'Invalid` or `Restricted` in the name?

Not necessary to do as part of this PR though.

Reply:
Considering there's no typed consumer of these exceptions, the fact that it's a parsing exception vs restricted or whatever we get to call it, is about irrelevant semantics.
Further more if the client sees the message wrapped (inside the UI for example), the exception type is even further minimized.
My point is, the message sent to the user is important, not the exception in which is wrapped.